compiler/ast/src/main/kotlin/com/uber/xprocessing/ext/XType.kt

```diff
@@ -312,8 +312,16 @@ fun XType.isPrimitive(): Boolean {
 }
 
 private fun XType.hasCollectionType(): Boolean {
-  return this.typeElement?.name.orEmpty() in collectionTypes ||
-      typeArguments.any { it.hasCollectionType() }
+  val visited = mutableSetOf<XType>()
```

Comment:
Was our xprocessing copied out of Room? If so, do we need to keep this up to date with the upstream version and a fix is needed there (or is already fixed and we should pull it)?

Reply:
I'm not sure if we copied out of Room. I admit I'm opening a PR here without much context on the big picture, history etc. 

@jbarr21 would you know?

Reply:
We consume xprocessing from Google's maven repo, though there are likely newer versions available 
https://github.com/uber/motif/blob/main/gradle/dependencies.gradle#L28